app/main.py

```diff
@@ -1,25 +1,72 @@
+from __future__ import annotations
+from typing import Any
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            obj: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> Any:
+        return getattr(obj, self.protected_name)
+
+    def __set__(self, obj: SlideLimitationValidator, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(obj, self.protected_name, value)
+        else:
+            raise ValueError(f"Value should not be less than {self.min_amount}"
+                             f"and greater than {self.max_amount}")
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, instance: Visitor) -> None:
+        try:
+            self.limitation_class(instance.age,
+                                  instance.weight,
+                                  instance.height)
```

Comment:
Read this message please
https://mate.academy/chats/244801?selectedMessageId=3129519